meorg_client/tests/test_client.py

```diff
@@ -62,38 +105,74 @@ def client():
 def test_login(client, mocker):
     """Test login."""
 
+    request = REQUEST_EXAMPLES['/login']
+    response = RESPONSE_EXAMPLES['/login']
+
     with mocker as rm:
-        client.login(**LOGIN_REQUEST)
+        client.login(**request)
 
         # Ensure the headers indicating successful login are set.
-        assert client.headers['X-User-Id'] == LOGIN_REPONSE['userId']
-        assert client.headers['X-Auth-Token'] == LOGIN_REPONSE['authToken']
+        assert client.headers['X-User-Id'] == response['userId']
+        assert client.headers['X-Auth-Token'] == response['authToken']
 
 
 def test_logout(client, mocker):
     """Test logout."""
 
+    request = REQUEST_EXAMPLES['/login']
+    response = RESPONSE_EXAMPLES['/logout']
+
     with mocker as rm: 
-        client.login(**LOGIN_REQUEST)
+        client.login(**request)
         client.logout()
 
         # Ensure the headers are cleared, indicating logout.
         assert client.headers == dict()
 
 
+def test_upload(client, mocker):
+    """Test upload."""
+
+    # Just use the api spec as the file to upload, it doesn't matter
+    filepath = get_installed_data_root() / 'openapi.json'
+    
+    with mocker as rm:
+        client.login(**REQUEST_EXAMPLES['/login'])
+        response = client.upload_file(filepath)
+        assert response.status_code == 200
```

Comment:
This assumes `upload_file` returns a `response` object. From the type hint, `upload_file()` can also return a dictionary.

I think I mentioned this previously but down the line I think we should handle request objects in the client. Is this worth doing now since we are writing tests?

Reply:
This is a sticking point, I do agree that the client should handle the response object and return a dict, but as you can see not everything returns any data, either only a return code or an unsuccessful response - so there are arguments for both.

I think there are probably 2 things to consider here:
1. The client should maintain an attribute for the most recent return code.
2. The client should raise an exception for unsuccessful response.

I am partial to using both approaches - leaving it up to dependent code to handle any exceptions.

Thoughts?

Reply:
I think would raise an exception in the client if the request is unsuccessful i.e. response is not 200.